src/languages/php.md

```diff
@@ -140,6 +140,50 @@ This is the complete list of extensions that can be enabled:
 > see the up-to-date complete list of extensions after you SSH into
 > your environment. For PHP 7, use `ls /etc/php/*/mods-available`.
 
+## Custom PHP extensions
+
+It is possible to use an extension not listed here, but there are a few steps involved.
+
+1. Download the .so file for the extension as part of your build hook using `curl` or similar.  It can also be added to your Git repository if the file is not publicly downloadable, although committing large binary blobs to Git is generally not recommended.
+
+2. Provide a custom `php.ini` file in the application root (as a sibling of your `.platform.app.yaml` file) that loads the extension using an absolute path.  For example, if the extension is named `spiffy.so` and is in the root of your application, you would have a `php.ini` file that reads:
+
+```ini
+extension=/app/spiffy.so
+```
+
+### Extensions needed by Composer
+
+In a few very rare cases, an extension may be needed by Composer itself, due to a Composer plugin depending on it.  If so, there's a few extra steps that are needed.
+
+3. Update your build flavor to none in your `.platform.app.yaml` file. You will need to provide a custom Composer command to use the extra extension. 
+
+```yaml
+build:
+    flavor: none
+```
+
+4. Add `PHP_INI_SCAN_DIR` environment variable and run `composer install` in your build hook.
+
+```yaml
+hooks:
+    build: |
+        export PHP_INI_SCAN_DIR="$(pwd):"
+        composer --no-ansi --no-interaction install --no-progress --prefer-dist --optimize-autoloader
+```
+
+If your extension is not in the application root then modify the directory accordingly.  (For instance, `export PHP_INI_SCAN_DIR="$(pwd)/my_extensions:"`)
```

Comment:
Let's simplify things and remove this paragraph by assuming the extension is always on the application root?